src/AppInstallerCommonCore/Public/winget/ManifestCommon.h

```diff
@@ -135,16 +135,36 @@ namespace AppInstaller::Manifest
     {
         DependencyList() = default;
 
-        void Add(const Dependency& dependency) 
+        void Add(const Dependency& newDependency)
         { 
-            dependencies.push_back(dependency); 
+            Dependency* existingDependency = this->HasDependency(newDependency);
+
+            if (existingDependency != NULL) {
+                if (newDependency.MinVersion) 
+                {
+                    if (existingDependency->MinVersion)
+                    {
+                        const auto& newDependencyVersion = AppInstaller::Utility::Version(newDependency.MinVersion.value());
+                        const auto& existingDependencyVersion = AppInstaller::Utility::Version(existingDependency->MinVersion.value());
+                        existingDependency->MinVersion.value() = newDependencyVersion <= existingDependencyVersion ? existingDependencyVersion.ToString() : newDependencyVersion.ToString();

```

Comment:
Given that there are allocations required, I would structure this as "write only if needed":
```
if (existing < new)
  existing->MV = new.MV;
```